internal/shared/api/handlers/handler/supply.go

```diff
@@ -0,0 +1,130 @@
+package handler
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"net/http"
+	"time"
+)
+
+// IntegrationSupply handler returns supply information to an external client (e.g., CoinMarketCap).
+// Note that the error text is returned to the client, so avoid including sensitive data in errors.
+func (h *Handler) IntegrationSupply(req *http.Request) (any, error) {
+	supplyType := req.URL.Query().Get("type")
+	switch supplyType {
+	case "total":
+		return h.totalSupply(req)
+	case "circulation":
+		return h.circulationSupply(req)
+	default:
+		return nil, fmt.Errorf("wrong type parameter='%s' (please provider either 'total' or 'circulation')", supplyType)
+	}
+}
+
+func (h *Handler) totalSupply(req *http.Request) (any, error) {
+	if h.bbnClient == nil {
+		return nil, fmt.Errorf("bbn configuration is not set")
+	}
+	ctx := req.Context()
+
+	coin, err := h.bbnClient.GetTotalSupply(ctx)
+	if err != nil {
+		log.Ctx(ctx).Err(err).Msg("Failed to get total supply")
+		return nil, fmt.Errorf("internal error")
+	}
+
+	return coin.Amount.Uint64(), nil
```

Comment:
I think we should ask them btw how would they treat this number - as json number or something else. AFAIK some js parsers won't handle large numbers and it's safer to return them as strings